src/libs/NextStepUtils.ts

```diff
@@ -27,5 +48,268 @@ function parseMessage(messages: Message[] | undefined) {
     return `<next-step>${formattedHtml}</next-step>`;
 }
 
-// eslint-disable-next-line import/prefer-default-export
-export {parseMessage};
+type BuildNextStepParameters = {
+    isPaidWithWallet?: boolean;
+};
+
+/**
+ * Generates an optimistic nextStep based on a current report status and other properties.
+ *
+ * @param report
+ * @param predictedNextStatus - a next expected status of the report
+ * @param parameters.isPaidWithWallet - Whether a report has been paid with the wallet or outside of Expensify
+ * @returns nextStep
+ */
+function buildNextStep(report: Report, predictedNextStatus: ValueOf<typeof CONST.REPORT.STATUS_NUM>, {isPaidWithWallet}: BuildNextStepParameters = {}): ReportNextStep | null {
+    const policy = ReportUtils.getPolicy(report.policyID ?? '');
+    const {submitsTo, isHarvestingEnabled, isPreventSelfApprovalEnabled, autoReportingFrequency, autoReportingOffset} = policy;
+    const {ownerAccountID = -1, managerID = -1} = report;
+    const isOwner = currentUserAccountID === ownerAccountID;
+    const isManager = currentUserAccountID === managerID;
+    const isSelfApproval = currentUserAccountID === submitsTo;
+    const ownerLogin = PersonalDetailsUtils.getLoginsByAccountIDs([ownerAccountID])[0] ?? '';
+    const submitterDisplayName = isSelfApproval ? 'you' : ReportUtils.getDisplayNameForParticipant(submitsTo, true) ?? '';
```

Comment:
The `submitsTo` is the manager of the report so the person who is in charge or reviewing and approving the report. 
```suggestion
    const managerDisplayName = isSelfApproval ? 'you' : ReportUtils.getDisplayNameForParticipant(submitsTo, true) ?? '';
```

Reply:
Done - 5b63982.